packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' as io;
+import 'dart:typed_data';
+
+import 'package:collection/collection.dart';
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:meta/meta.dart';
+import 'package:platform/platform.dart';
+import 'package:process/process.dart';
+
+const String _kFlutterRootKey = 'FLUTTER_ROOT';
+
+/// Main method that can be used in a `flutter_test_config.dart` file to set
+/// [goldenFileComparator] to an instance of [FlutterGoldenFileComparator] that
+/// works for the current test.
+Future<void> main(FutureOr<void> testMain()) async {
+  goldenFileComparator = await FlutterGoldenFileComparator.fromDefaultComparator();
+  await testMain();
+}
+
+/// A golden file comparator specific to the `flutter/flutter` repository.
+///
+/// Within the https://github.com/flutter/flutter repository, it's important
+/// not to check-in binaries in order to keep the size of the repository to a
+/// minimum. To satisfy this requirement, this comparator retrieves the golden
+/// files from a sibling repository, `flutter/goldens`.
+///
+/// This comparator will locally clone the `flutter/goldens` repository into
+/// the `$FLUTTER_ROOT/bin/cache/pkg/goldens` folder, then perform the comparison against
+/// the files therein.
+class FlutterGoldenFileComparator implements GoldenFileComparator {
+  @visibleForTesting
+  FlutterGoldenFileComparator(
+    this.goldens,
+    this.basedir, {
+    this.fs: const LocalFileSystem(),
+  });
+
+  final GoldensClient goldens;
+  final Uri basedir;
+  final FileSystem fs;
+
+  /// Creates a new [FlutterGoldenFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// By the time the future completes, the clone of the `flutter/goldens`
+  /// repository is guaranteed to be ready use.
+  static Future<FlutterGoldenFileComparator> fromDefaultComparator() async {
+    final LocalFileComparator defaultComparator = goldenFileComparator;
+    final GoldensClient goldens = new GoldensClient();
+    await goldens.prepare();
+    return FlutterGoldenFileComparator(goldens, defaultComparator.basedir);
```

Comment:
let's keep using "new" until we have a lint to tell us not to

Reply:
cc @a14n

Reply:
Done